lib/core/stores/local.ts

```diff
@@ -0,0 +1,364 @@
+import {
+  DurablePromise,
+  PendingPromise,
+  ResolvedPromise,
+  RejectedPromise,
+  CanceledPromise,
+  TimedoutPromise,
+  isPendingPromise,
+  isResolvedPromise,
+  isRejectedPromise,
+  isCanceledPromise,
+} from "../promise";
+import { IPromiseStore } from "../store";
+import { IEncoder } from "../encoder";
+import { Base64Encoder } from "../encoders/base64";
+import { ErrorCodes, ResonateError } from "../error";
+
+export class LocalPromiseStore implements IPromiseStore {
+  private readonly storeName = "promises";
+  private db: IDBDatabase;
+
+  private constructor(
+    db: IDBDatabase,
+    private encoder: IEncoder<string, string> = new Base64Encoder(),
+  ) {
+    this.db = db;
+  }
+
+  static async getInstance(encoder?: IEncoder<string, string>): Promise<LocalPromiseStore> {
+    return new Promise<LocalPromiseStore>((resolve, reject) => {
+      const request = indexedDB.open("resonateDB", 1);
+
+      request.onerror = () => {
+        reject(request.error);
+      };
+
+      request.onsuccess = () => {
+        const db = request.result;
+        resolve(new LocalPromiseStore(db, encoder));
+      };
+
+      request.onupgradeneeded = (event) => {
+        const db = (event.target as IDBOpenDBRequest).result;
+        db.createObjectStore("promises", { keyPath: "id" });
+      };
+    });
+  }
+
+  private async getObjectStore(): Promise<IDBObjectStore> {
+    if (!this.db) {
+      throw new ResonateError("Database not initialized", ErrorCodes.DATABASE);
+    }
+
+    const transaction = this.db.transaction(this.storeName, "readwrite");
```

Comment:
Do we need to call commit on the transaction? [the docs](https://developer.mozilla.org/en-US/docs/Web/API/IDBTransaction/commit) are kinda ambiguous

Reply:
It says, commits happen automatically. 

The commit happens implicitly when the promise returned by the transaction method is resolved.